gateway/tests/tests/tests.rs

```diff
@@ -44,13 +44,26 @@ async fn test_gateway_authentication() -> Result<()> {
     // Create an RPC client
     let client = RpcClient::new(gw_announce_address);
 
-    // Register a test federation
+    // Test gateway authentication on `register_federation` function
+    // *  `register_federation` with correct password succeeds
+    // *  `register_federation` with incorrect password fails
     let payload = RegisterFedPayload {
         connect: serde_json::to_string(&WsFederationConnect { members: vec![] })?,
     };
-    client
-        .register_federation(gw_password.clone(), payload)
-        .await?;
+    assert_eq!(
+        client
+            .register_federation("registerfed".to_string(), payload.clone())
+            .await?
+            .status(),
+        401
+    );
+    assert_ne!(
+        client
+            .register_federation(gw_password.clone(), payload)
+            .await?
+            .status(),
+        401
+    );
```

Comment:
Wondering if we can make this a generic function or macro, e.g.

```rust
fn test_auth<T>(client: &Client, func: fn(&Client, String, T) -> impl Future<…>);
```

Reply:
Pending: I'll have to think more about this one